.ci/README.md

```diff
@@ -1,63 +1,87 @@
+* Tasks are things which get run to produce state internal to Concourse.  A shell script, running inside a container, with pre-specified input and output directories, is a task.  A task can have many inputs and outputs.  The inputs are usually state specified by a resource version, or an output of a previous task.  The outputs are usually used to update resource versions, or as inputs to a later task.
+* Jobs tie together Resources and Tasks.  One task that we have can be summarized as "run the MagicModules compiler to generate Terraform, and output the resulting Terraform directory.  The associated Job is much more complicated - it can be summarized as:
+  * Download a version of the MagicModules PR Resource.
+  * Run it through one task to prepare it.
+  * Run it through the task to generate Terraform.
+  * Upload the output to the Terraform Resource as a new version.
+  * Run the task to cause the output MagicModules repository's Terraform submodule to point to the new Terraform version.
+  * Upload the output MagicModules repository as a new version to the MagicModules Resource.
+
+## Containers
+
+Concourse is strongly based on containers, and consequently it's worth keeping track of which containers the pipeline uses.
+
+### nmckinley/concourse-github-pr-resource
+This is based on https://github.com/jtarchie/github-pullrequest-resource.  It contains a few features that haven't been merged into that repo yet, which you can see at this fork: https://github.com/ndmckinley/github-pullrequest-resource.  It is probably not necessary to update this container for now, and we should switch to using jtarchie/concourse-github-pr-resource ASAP, but if there are problems in the meantime, message @ndmckinley.
+
+### nmckinley/concourse-git-resource
+This is based on https://github.com/concourse/git-resource/pull/172.  This PR can't be accepted into that repository as-is, because it makes drastic changes, but the proposed replacements aren't ready yet.  This container will also go away when those changes are made.  It shouldn't be necessary to update, but if, for instance, you need a feature which was added to the git resource after the container was built, message @ndmckinley.
+
+-------
+
+#### A side note on tracking submodules
+The merge job sets the submodules back to tracking their downstream repositories on `master`.  It updates the submodules to point to the most recent commit on `master`.  This may not be ideal - other commits may have been made to `master` since the downstream PR was merged - however, this is the best way to ensure that we do not go backwards.  Imagine the following situation:
```

Comment:
optional nit- suggested phrasing: "The merge job sets the submodules back to tracking their downstream repositories by  updating the submodules to point to the most recent commit on `master`"

Reply:
Done.